src/dispatch/plugins/dispatch_slack/case/messages.py

```diff
@@ -0,0 +1,180 @@
+from typing import List
+from blockkit import Actions, Button, Context, Message, Section, Divider
+
+from dispatch.config import DISPATCH_UI_URL
+from dispatch.case.enums import CaseStatus
+from dispatch.case.models import Case
+from dispatch.plugins.dispatch_slack.models import SubjectMetadata
+from dispatch.plugins.dispatch_slack.case.enums import (
+    CaseNotificationActions,
+)
+from dispatch.plugins.dispatch_slack.service import chunks
+
+
+def create_case_message(case: Case, channel_id: str):
+    blocks = [
+        Context(elements=["*Case Details*"]),
+        Section(
+            text=f"*Title* \n {case.title}.",
+            accessory=Button(
+                text="View",
+                url=f"{DISPATCH_UI_URL}/{case.project.organization.slug}/cases/{case.name}",
+            ),
+        ),
+        Section(
+            text=f"*Description* \n {case.description} \n \n Additional information is available in the <{case.case_document.weblink}|case document>."
+        ),
+        Section(
+            fields=[
+                f"*Assignee* \n {case.assignee.email}",
+                f"*Status* \n {case.status}",
+                f"*Severity* \n {case.case_severity.name}",
+                f"*Type* \n {case.case_type.name}",
+                f"*Priority* \n {case.case_priority.name}",
+            ]
+        ),
```

Comment:
```suggestion
        ),
        Divider(),
```

Reply:
Made a mockup of a case notification block with some extended options. Not all required for MVP. But, sharing in case you like any differences.

Here's a block kid builder view: [link](https://app.slack.com/block-kit-builder/T04FZTFLBFE#%7B%22blocks%22:%5B%7B%22type%22:%22context%22,%22elements%22:%5B%7B%22type%22:%22plain_text%22,%22text%22:%22%F0%9F%92%A1%20New%20Case%20Notification%22,%22emoji%22:true%7D%5D%7D,%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Title%22,%22emoji%22:true%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22This%20is%20a%20mrkdwn%20section%20block%22%7D%7D,%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Description%22,%22emoji%22:true%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22It%20is%20a%20long%20established%20fact%20that%20a%20reader%20will%20be%20distracted%20by%20the%20readable%20content%20of%20a%20page%20when%20looking%20at%20its%20layout.%20The%20point%20of%20using%20Lorem%20Ipsum%20is%20that%20it%20has%20a%20more-or-less%20normal%20distribution%20of%20letters,%20as%20opposed%20to%20using%20'Content%20here,%20content%20here',%20making%20it%20look%20like%20readable%20English.%20Many%20desktop%20publishing%20packages%20and%20web%20page%20editors%20now%20use%20Lorem%20Ipsum%20as%20their%20default%20model%20text,%20and%20a%20search%20for%20'lorem%20ipsum'%20will%20uncover%20many%20web%20sites%20still%20in%20their%20infancy.%22%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Assignee*%20%22%7D,%22accessory%22:%7B%22type%22:%22users_select%22,%22placeholder%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Select%20a%20user%22,%22emoji%22:true%7D,%22action_id%22:%22users_select-action%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Related%20Incidents*%22%7D,%22accessory%22:%7B%22type%22:%22multi_conversations_select%22,%22placeholder%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Select%20incidents%22,%22emoji%22:true%7D,%22action_id%22:%22multi_conversations_select-action%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Related%20Cases*%22%7D,%22accessory%22:%7B%22type%22:%22multi_static_select%22,%22placeholder%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Select%20cases%22,%22emoji%22:true%7D,%22options%22:%5B%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22*this%20is%20plain_text%20text*%22,%22emoji%22:true%7D,%22value%22:%22value-0%22%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22*this%20is%20plain_text%20text*%22,%22emoji%22:true%7D,%22value%22:%22value-1%22%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22*this%20is%20plain_text%20text*%22,%22emoji%22:true%7D,%22value%22:%22value-2%22%7D%5D,%22action_id%22:%22multi_static_select-action%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Tags*%22%7D,%22accessory%22:%7B%22type%22:%22multi_static_select%22,%22placeholder%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Select%20tags%22,%22emoji%22:true%7D,%22options%22:%5B%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22*this%20is%20plain_text%20text*%22,%22emoji%22:true%7D,%22value%22:%22value-0%22%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22*this%20is%20plain_text%20text*%22,%22emoji%22:true%7D,%22value%22:%22value-1%22%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22*this%20is%20plain_text%20text*%22,%22emoji%22:true%7D,%22value%22:%22value-2%22%7D%5D,%22action_id%22:%22multi_static_select-action%22%7D%7D,%7B%22type%22:%22section%22,%22fields%22:%5B%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Priority*%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Status*%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22%60Low%60%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22%60New%60%22%7D%5D%7D,%7B%22type%22:%22section%22,%22fields%22:%5B%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Severity*%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22*Type*%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22%60Medium%60%22%7D,%7B%22type%22:%22mrkdwn%22,%22text%22:%22%60Security%20Triage%20and%20Analysis%60%22%7D%5D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22%20%22,%22emoji%22:true%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22actions%22,%22elements%22:%5B%7B%22type%22:%22button%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Edit%22,%22emoji%22:true%7D,%22value%22:%22click_me_123%22,%22action_id%22:%22actionId-0%22%7D,%7B%22type%22:%22button%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Resolve%22,%22emoji%22:true%7D,%22style%22:%22primary%22,%22value%22:%22click_me_123%22,%22action_id%22:%22actionId-1%22%7D,%7B%22type%22:%22button%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22Escalate%22,%22emoji%22:true%7D,%22style%22:%22danger%22,%22value%22:%22click_me_123%22,%22action_id%22:%22actionId-2%22%7D,%7B%22type%22:%22overflow%22,%22options%22:%5B%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22%F0%9F%91%80%20View%22,%22emoji%22:true%7D,%22url%22:%22https://google.com%22,%22value%22:%22value-0%22%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22%F0%9F%8E%AB%20Ticket%22,%22emoji%22:true%7D,%22url%22:%22https://google.com%22,%22value%22:%22value-2%22%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22%F0%9F%93%A6%20Storage%22,%22emoji%22:true%7D,%22url%22:%22https://google.com%22,%22value%22:%22value-3%22%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22%F0%9F%93%93%20Notebook%22,%22emoji%22:true%7D,%22url%22:%22https://google.com%22,%22value%22:%22value-3%22%7D,%7B%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22%F0%9F%93%84%20Document%22,%22emoji%22:true%7D,%22url%22:%22https://google.com%22,%22value%22:%22value-1%22%7D%5D,%22action_id%22:%22overflow-action%22%7D%5D%7D%5D%7D)

```python
from blockkit import (
    Actions,
    Button,
    Context,
    Divider,
    Header,
    Message,
    MultiConversationsSelect,
    MultiStaticSelect,
    Overflow,
    PlainOption,
    Section,
    UsersSelect,
)

payload = Message(
    blocks=[
        Context(elements=["💡 New Case Notification"]),
        Header(text="Title"),
        Section(text="This is a mrkdwn section block"),
        Header(text="Description"),
        Section(
            text="It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for 'lorem ipsum' will uncover many web sites still in their infancy."
        ),
        Divider(),
        Section(
            text="*Assignee* ",
            accessory=UsersSelect(
                placeholder="Select a user", action_id="users_select-action"
            ),
        ),
        Section(
            text="*Related Incidents*",
            accessory=MultiConversationsSelect(
                placeholder="Select incidents",
                action_id="multi_conversations_select-action",
            ),
        ),
        Section(
            text="*Related Cases*",
            accessory=MultiStaticSelect(
                placeholder="Select cases",
                options=[
                    PlainOption(text="*this is plain_text text*", value="value-0"),
                    PlainOption(text="*this is plain_text text*", value="value-1"),
                    PlainOption(text="*this is plain_text text*", value="value-2"),
                ],
                action_id="multi_static_select-action",
            ),
        ),
        Section(
            text="*Tags*",
            accessory=MultiStaticSelect(
                placeholder="Select tags",
                options=[
                    PlainOption(text="*this is plain_text text*", value="value-0"),
                    PlainOption(text="*this is plain_text text*", value="value-1"),
                    PlainOption(text="*this is plain_text text*", value="value-2"),
                ],
                action_id="multi_static_select-action",
            ),
        ),
        Section(fields=["*Priority*", "*Status*", "`Low`", "`New`"]),
        Section(
            fields=[
                "*Severity*",
                "*Type*",
                "`Medium`",
                "`Security Triage and Analysis`",
            ]
        ),
        Section(text=" "),
        Divider(),
        Actions(
            elements=[
                Button(text="Edit", value="click_me_123", action_id="actionId-0"),
                Button(
                    text="Resolve",
                    style="primary",
                    value="click_me_123",
                    action_id="actionId-1",
                ),
                Button(
                    text="Escalate",
                    style="danger",
                    value="click_me_123",
                    action_id="actionId-2",
                ),
                Overflow(
                    options=[
                        PlainOption(
                            text="👀 View", url="https://google.com", value="value-0"
                        ),
                        PlainOption(
                            text="🎫 Ticket", url="https://google.com", value="value-2"
                        ),
                        PlainOption(
                            text="📦 Storage", url="https://google.com", value="value-3"
                        ),
                        PlainOption(
                            text="📓 Notebook", url="https://google.com", value="value-3"
                        ),
                        PlainOption(
                            text="📄 Document", url="https://google.com", value="value-1"
                        ),
                    ],
                    action_id="overflow-action",
                ),
            ]
        ),
    ]
).build()
```

Reply:
I do like several of these changes. I'll circle back and walk through them without.